ProcessOut/Classes/ProcessOut.swift

```diff
@@ -550,8 +604,69 @@ public class ProcessOut {
             handler.onError(error: ProcessOutException.InternalError)
         }
     }
-    
-    
+
+    /// Create a customer token from a card ID
+    ///
+    /// - Parameters:
+    ///   - TokenRequest: contains all the fields necessary for the token request
+    ///   - handler: 3DS2 handler
+    ///   - with: UIViewController to display webviews and perform fingerprinting
+    public static func makeCardToken(TokenRequest: TokenRequest, handler: ThreeDSHandler, with: UIViewController) {
+        // Handle empty source
+        if (TokenRequest.getSource().isEmpty()) {
+            handler.onError(error: ProcessOutException.InternalError)
+            return
+        }
+
+        guard let body = try? JSONEncoder().encode(TokenRequest) else {
+            handler.onError(error: ProcessOutException.InternalError)
+            return
+        }
+        do {
+            let json = try JSONSerialization.jsonObject(with: body, options: []) as! [String: Any]
+            HttpRequest(route: "/customers/" + customerId + "/tokens/" + tokenId, method: .put, parameters: json) { (data, error) in
```

Comment:
`customerId` and `tokenId` will be coming now from the TokenRequest, you have to update that here.